core/src/main/scala/scala/pickling/util/ClassMapper.scala

```diff
@@ -0,0 +1,33 @@
+package scala.pickling.util
+
+object ClassMapper {
+
+  val specializedTuplesTemplate =
+    List(
+      "$mcII$sp", "$mcIJ$sp", "$mcID$sp", "$mcIZ$sp", "$mcJI$sp", "$mcJJ$sp",
+      "$mcJD$sp", "$mcJC$sp", "$mcJZ$sp", "$mcDI$sp", "$mcDJ$sp", "$mcDD$sp",
+      "$mcDC$sp", "$mcDZ$sp", "$mcCI$sp", "$mcCJ$sp", "$mcCD$sp", "$mcCC$sp",
+      "$mcCZ$sp", "$mcZI$sp", "$mcZJ$sp", "$mcZD$sp", "$mcZC$sp", "$mcZZ$sp"
+    )
+
+  /* Map specialized classes to classes. Canonical use case: tuples.
+   * We map classes instead of strings to check at runtime that they exist. */
+  val specialMappingClasses: Map[Class[_], Class[_]] =
+    mapSpecializedTuplesFor("scala.Tuple2") // add also other special cases
```

Comment:
We may be able to make this more generic, i forget the rules fo r specialization mangling